pkg/platform/runtime/target/target.go

```diff
@@ -152,13 +162,60 @@ func (c *CustomTarget) Headless() bool {
 	return c.headless
 }
 
-func ProjectDirToTargetDir(projectDir, cacheDir string) string {
-	resolvedDir, err := fileutils.ResolveUniquePath(projectDir)
+func (c *CustomTarget) ReadOnly() bool {
+	return c.commitUUID == ""
+}
+
+func (c *CustomTarget) InstallFromDir() *string {
+	return nil
+}
+
+type InstallationTarget struct {
+	owner        string
+	name         string
+	commitUUID   strfmt.UUID
+	dir          string
+	artifactsDir string
+}
+
+func NewInstallationTarget(owner string, name string, commitUUID strfmt.UUID, dir string, artifactsDir string) *InstallationTarget {
+	cleanDir, err := fileutils.ResolveUniquePath(dir)
 	if err != nil {
-		multilog.Error("Could not resolve unique path for projectDir: %s, error: %s", projectDir, err.Error())
-		resolvedDir = projectDir
+		multilog.Error("Could not resolve unique path for dir: %s, error: %s", dir, err.Error())
+	} else {
+		dir = cleanDir
 	}
-	logging.Debug("In newStore: resolved project dir is: %s", resolvedDir)
+	return &InstallationTarget{owner, name, commitUUID, dir, artifactsDir}
+}
 
-	return filepath.Join(cacheDir, hash.ShortHash(resolvedDir))
+func (i *InstallationTarget) Owner() string {
+	return i.owner
+}
+
+func (i *InstallationTarget) Name() string {
+	return i.name
+}
+
+func (i *InstallationTarget) CommitUUID() strfmt.UUID {
+	return i.commitUUID
+}
+
+func (i *InstallationTarget) Dir() string {
+	return i.dir
+}
+
+func (i *InstallationTarget) Trigger() Trigger {
+	return triggerUnknown
```

Comment:
`triggerUnknown` is an error case, effectively. It's saying "we tried but we can't figure it out". I don't think that's warranted here. We should introduce a new trigger maybe, eg. "offline".